### PR TITLE
Make combine_results fail rather than stall upon exception with Dask

### PR DESCRIPTION
**Pull request by lathanh**
**Background**: `dask.compute(map(concat_partial, ...` (in `postprocessing.combine_results()`) can fail[1], and currently, such an exception is somehow handled in a way that doesn't kill the script. Instead, the script hangs until the Cloud Run job times out (currently 24h).

Also, this error can occur non-deterministically. Given the particular configuration that I was testing with, rerunning the same exact command/config over and over fails about 30% of the time.

**Change**: Catch the exception and fail fast (i.e., call `sys.exit(1)`).

[1] With a error like "_distributed.scheduler.KilledWorker: Attempted to run task concat_and_normalize-56525f85-b55a-49e9-97bf-bc2bf253757e on 3 different workers, but all those workers died while running it. The last worker that attempt to run the task was tcp://127.0.0.1:43341. Inspecting worker logs is often a good next step to diagnose what went wrong. For more information see https://distributed.dask.org/en/stable/killed.html._"

### CODE REVIEW

File: buildstockbatch/postprocessing.py [hunk 1]
```diff
@@ -597,13 +598,17 @@ def combine_results(fs, results_dir, cfg, do_timeseries=True):
...
+                    try:
+                        dask.compute(
+                            map(
+                                concat_partial,
+                                *zip(*enumerate(bldg_id_groups)),
+                                partition_vals_list,
+                            )
                         )
-                    )
+                    except:
```
Comment (nweires) [inline buildstockbatch/postprocessing.py:609]: Do we always see the same type of error here? If so, we should specify it (`except KilledWorker`) so we don't accidentally catch other error types.

**Review by nweires — APPROVED** [review]
A bit of poking around makes me wonder if this is a Dask issue. (Something like [this](https://stackoverflow.com/questions/66052839/terminating-dask-workers-after-jobs-are-done), maybe?)

File: buildstockbatch/postprocessing.py (hunk 1 above)
Comment (lathanh) [reply buildstockbatch/postprocessing.py:609]: I've only ever seen KilledWorker. However, I don't see why any other exception would have a different result; i.e., they would also cause a stall. I think it's better to have this case be flexible to also handle the unknown-unknown (possible other exceptions) than to let those cause a stall (and someone has to re-debug what's happening).

So, resolving and submitting as-is, but feel free to continue the conversation and we can change in a follow-up.

**Comment by lathanh** [comment]
> A bit of poking around makes me wonder if this is a Dask issue. (Something like [this](https://stackoverflow.com/questions/66052839/terminating-dask-workers-after-jobs-are-done), maybe?)

I think both... I think there's a Dask issue in that it can stall like that, but there's also an issue at a higher level with the exception getting eaten. I'm not sure how to solve either (yet), but I at least want to get something in that addresses the impact of the issues.

**Pull request merged**